source/logrepl/snapshot.go

```diff
@@ -133,7 +141,15 @@ func (s *SnapshotIterator) Teardown(ctx context.Context) error {
 	if commitErr := s.tx.Commit(ctx); commitErr != nil {
```

Comment:
Might be a dumb question, but why do we call s.tx.Commit() here at all?

Reply:
Simply to end the transaction.

Reply:
Ah, I think I need a break.:D I was thinking this is the source teardown, but it's the iterator teardown, and actually a different thing in mind because of that.